File: Nivel1/Estadistica/Scripts/CalculadoraEstadisticaBasica.R
get_tamano_muestra <- function(){
    cat('\n----- OBTENER TAMAÑO DE MUESTRA -----\n\n[1]Población finita.\n[2]Población infinita\n[3]Desviación estandar.\n[s]Salir.\n')
    opc <- readline(prompt = 'Ingresa una opción: ')

    if(opc %in% c('1','2','3')){
        while(TRUE){
            tryCatch({
                n <- 0
                z <- as.numeric(readline(prompt = '\nIngrese el valor del nivel de confianza: '))
                p <- as.numeric(readline(prompt = 'Ingrese la proporción esperada de la población [entre 0 y 1]: '))
                e <- as.numeric(readline(prompt = 'Ingrese el margen de error[entre 0 y 1]: '))
                q <- 1 - p

                if(opc == '1'){
                    poblacion <- as.numeric(readline(prompt = 'Ingrese la población: '))
                    n = (poblacion * z ^ 2 * p * q) / (e ^ 2 * (poblacion - 1) + z ^ 2 * p * q)
                }else{
                    if(opc == '2'){n = (z ^ 2 * p * q) / e ^ 2}else{n = 1}
                }

                cat('\n\nEl tamaño de la muestra es de: ', n, ' --> ', round(n), '.\n\n')
                break

            }, error = function(e){
                if(readline(prompt = '\nERROR!, debe ser un dato numérico; desea repetir el proceso: [s/n]: ') != 's'){break}
            })
        }
    }
}

get_tamano_muestra()

File: Nivel1/Estadistica/Scripts/calculadora_estadistica_basica.py
# Operaciones: -----------------------------------------------------------------

def obt_tamano_muestra(opcion, N, z, p, e, s):
    if opcion == "1":
        resultado = (N * z**2 * p * (1 - p)) / (e**2 * (N - 1) + z**2 * p * (1 - p))
    elif opcion == "2":
        resultado = (z**2 * p * (1 - p)) / e**2
    else:
         resultado = (z * s / e)**2
    return resultado


# Validaciones: ----------------------------------------------------------------

def obt_respuesta_numerica(pregunta):
    while True:
        entrada = input(pregunta)
        try:
            respuesta = float(entrada)
            return respuesta
        except ValueError:
            print("\nError...")


# Opciones de la calculadora: --------------------------------------------------

def imp_tamano_muestra():
    print("\n------ OBTENER TAMAÑO DE MUESTRA ------\n")
    print("[1] Con población finita.\n[2] Con población infinita\n[3] Con Desviación estandar.")
    print("Presiona cualquier otra letra para salir.")

    opcion = input("Ingresa una de la opciones: ")

    if opcion == "1" or opcion == "2" or opcion == "3":
        N = 0
        s = 0
        z = obt_respuesta_numerica("\nIngrese el valor del nivel de confianza: ")
        p = obt_respuesta_numerica("Ingrese la proporción esperada de la población [entre 0 y 1]: ")
        e = obt_respuesta_numerica("Ingrese el margen de error[entre 0 y 1]: ")

        if opcion == "1": 
            N = obt_respuesta_numerica("Ingrese el total de la población: ")
        if opcion == "3":
            s = obt_respuesta_numerica("Ingresa el valor de la desviación estandar: ")
        
        muestra = obt_tamano_muestra(opcion, N, z, p, e, s)

        print("\nEl tamaño de la muestra es de: ", muestra, " --> ", round(muestra), "\n")


# Función Principal: -----------------------------------------------------------

def main():
    imp_tamano_muestra()


main()